### PR TITLE
Feature/tool selector map

### PR DESCRIPTION
**Pull request by Bluesmile82**
This PR updates the world map with a tooltip, hover interactions and prepares the toolSelector map to have the interactions described in the design.

Missing:
- Some optimization (Fetch for the countries hover is slow sometimes)
- Animations

### CODE REVIEW

File: frontend/scripts/react-components/explore/explore.actions.js [hunk 1]
```diff
@@ -10,11 +10,11 @@ export const EXPLORE__SET_SELECTED_TABLE_COLUMN_TYPE = 'EXPLORE__SET_SELECTED_TA
 export const getTopNodesKey = (ctx, col, start, end) =>
   ctx && col && start && end ? `CTX${ctx}_COL${col}_START${start}_END${end}` : null;
 
-export const setExploreTopNodes = columnType => (dispatch, getState) => {
+export const setExploreTopNodes = (columnType, context) => (dispatch, getState) => {
   if (!columnType) return null;
 
   const state = getState();
-  const selectedContext = getSelectedContext(state);
+  const selectedContext = context || getSelectedContext(state);
```
Comment (sorodrigo) [inline frontend/scripts/react-components/explore/explore.actions.js]: I dont think this is necessary, your selectors should handle this for you

**Comment by agnessa** [comment]
just double checking that you'll still need an endpoint like this for the map flows: https://www.pivotaltracker.com/story/show/167758271

**Comment by Bluesmile82** [comment]
Yes, we do. Now it's working but it has to load the top nodes on request which is slow and it could give a wrong impression to the user. Instead, we could get them all loaded when we arrive at the page

File: frontend/scripts/react-components/shared/world-map/world-map.selectors.js [hunk 2]
```diff
@@ -6,8 +6,23 @@ import greatCircle from '@turf/great-circle';
...
+import { getContexts } from 'react-components/tool-selector/tool-selector.selectors';
+
+const getIsToolSelector = (state, { toolSelector }) => toolSelector;
+const getHighlightedContext = (state, { highlightedContext }) => highlightedContext;
+const getHighlightedCountryIds = (state, { highlightedCountryIds }) => highlightedCountryIds;
+const getSelectedContext = createSelector(
+  [getIsToolSelector, getHighlightedContext, getAppSelectedContext],
+  (isToolSelector, highlightedContext, appSelectedContext) => {
+    if (highlightedContext) return highlightedContext;
+    if (isToolSelector) return null;
```
[the remark this answers is not in the document]
Comment (Bluesmile82) [reply frontend/scripts/react-components/shared/world-map/world-map.selectors.js]: By default the WolrdMap component will show some flows but we only want to show the highlightedCountries on the first step if hovered.

[the remark this answers is not in the document]
Comment (sorodrigo) [reply frontend/scripts/react-components/shared/world-map/world-map.selectors.js]: oh then the world map should be agnostic to this detail. Maybe we're mixing explore selectors with world map selectors. If the world map is reused elsewhere, it should just care about what to render, without knowing where is being rendered.

**Comment by Bluesmile82** [comment]
image
There are some cases in which is difficult for the flows to not get cut

File: frontend/scripts/react-components/shared/world-map/world-map.selectors.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Bluesmile82) [reply frontend/scripts/react-components/shared/world-map/world-map.selectors.js]: Then we should extract also more logic to the home page. Currently, the map is finding out with context it should render from the store. We could have more props and less logic it's true

**Comment by sorodrigo** [comment]
I think it's possible, you just have to give a bigger height to the map. Even if that means having a bigger gap at the bottom of it.
Screen Shot 2019-08-21 at 10 11 24

**Pull request merged**